cmd/state-svc/main.go

```diff
@@ -70,6 +71,12 @@ func main() {
 	rollbar.SetupRollbar(constants.StateServiceRollbarToken)
 	rollbar.SetConfig(cfg)
 
+	if err := analytics.RegisterConfigListener(cfg); err != nil {
+		multilog.Critical("Could not register config listener: %v", errs.JoinMessage(err))
+		exitCode = 1
+		return
+	}
```

Comment:
Analytics is already using a single instance that is propagated everywhere, so there should be no need for globals. You can do the config listener registration in `analytics.New()`.

Reply:
I can't find a confirmation of this. There is no `analytics.New()`. We have multiple analytics clients and while the async client is created once and passed around throughout the state tool the sync client has multiple instantiations in the state service.

Reply:
In the state service we have two invocations of `sync.New()` one for the service itself and another for the resolver because I believe the latter is running in a goroutine. See [here](https://github.com/ActiveState/cli/blob/master/cmd/state-svc/main.go#L105) and [here](https://github.com/ActiveState/cli/blob/master/cmd/state-svc/internal/resolver/resolver.go#L88) also it looks like we're only using the pixel reporter in the sync client...